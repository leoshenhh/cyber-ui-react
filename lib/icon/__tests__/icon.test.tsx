import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });
import renderer from 'react-test-renderer'
import Icon from '../icon';
import { shallow , mount} from 'enzyme'

describe('button',()=>{
    it('是个div',()=>{
        const json = renderer.create(<Icon name="wechat"/>).toJSON()
        expect(json).toMatchSnapshot()
    })
    it('Test click event', () => {
        const mockCallBack = jest.fn();

        const component = shallow((<Icon name="wechat" onClick={mockCallBack}>Ok!</Icon>));
        component.find('svg').simulate('click');
        expect(mockCallBack.mock.calls.length).toEqual(1);
    });
    it('onClick', ()=>{
        const fn = jest.fn()
        const fn1 = jest.fn()
        const component = shallow(<Icon name="wechat" onClick={fn}/>)
        component.find('svg').simulate('click');
        expect(fn).toBeCalled()
        expect(fn1).toBeCalled()
    })
})
