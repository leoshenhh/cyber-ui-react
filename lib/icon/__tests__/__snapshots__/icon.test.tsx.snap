// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button 是个div 1`] = `
<svg
  className="icon"
>
  <use
    xlinkHref="#wechat"
  />
</svg>
`;
